### PR TITLE
test: Add Wallet Unlock Context Manager

### LINKED ISSUE #28601: test: unlocked wallet context manager
### Motivation

Currently tests with a locked wallet have to verbosely specify a long timeout and then lock the wallet again, once they are done testing the unlocked part. This is fragile, because the wallet may lock earlier, or the dev may forget to re-lock the wallet when done.

All issues should be fixed with a context manager.

### Possible solution

A context manager. See also https://github.com/bitcoin/bitcoin/pull/28403#discussion_r1325426430

### Useful Skills

* Compiling Bitcoin Core from source
* Running the C++ unit tests and the Python functional tests
* Python 3 contextlib

### Guidance for new contributors

Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by BrandonOdiwuor**
Fixes #28601, see https://github.com/bitcoin/bitcoin/pull/28403#discussion_r1325426430

Add Context Manager to manage the locking and unlocking of locked wallets with a passphrase during testing.

### CODE REVIEW

**Review by maflcko — APPROVED** [review]

**Review by maflcko — COMMENTED** [review]
I think there is one more that can be using the context:

```
test/functional/wallet_fundrawtransaction.py:        wallet.walletpassphrase("test", 999000)

File: test/functional/test_framework/wallet_util.py [hunk 1]
```diff
@@ -122,3 +122,21 @@ def generate_keypair(compressed=True, wif=False):
     if wif:
         privkey = bytes_to_wif(privkey.get_bytes(), compressed)
     return privkey, pubkey
+
+class WalletUnlockContextManager():
+    """
+    A context manager for unlocking a wallet with a passphrase and automatically locking it afterward.
+    """
+
+    TIMEOUT = 999000 # WALLET_PASSPHRASE_TIMEOUT
```
[the remark this answers is not in the document]
Comment (BrandonOdiwuor) [reply test/functional/test_framework/wallet_util.py]: done

**Comment by BrandonOdiwuor** [comment]
@maflcko could you please review the changes

**Comment by kevkevinpal** [comment]
lgtm ACK [004903e](https://github.com/bitcoin/bitcoin/pull/28617/commits/004903ebade38ba47c5ddc17b756d605b963528e)

**Comment by kashifs** [comment]
Hey @BrandonOdiwuor, nice work on these changes. I know that this PR has been closed, but I finally finished my review of all the files. Apologies for the delay. I'm trying to get up to speed and have to work very slowly at the moment. I learned a ton from your work and have already gotten some value from the code review, but I'm not sure of the best way to contribute these changes. They can be found here: 

https://github.com/bitcoin/bitcoin/compare/master...kashifs:bitcoin:master

Please let me know what you think. Thanks!

**Pull request merged**